### PR TITLE
FOUR-8284: Create a Helper to compare two arrays

### PR DESCRIPTION
**Pull request by CarliPinell**
## Issue & Reproduction Steps

Create a Helper to compare two arrays changes vs origin.
Event UserUpdate was changed to avoid keys from Package in User Information [package-advanced-user-manager](https://github.com/ProcessMaker/package-advanced-user-manager)

ArrayHelper
UserUpdated

The Helper has 2 methods:
stdClassToArray
getArrayDifferencesWithFormat

## Related Tickets & Packages
- FOUR-8284

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy

### CODE REVIEW

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

File: ProcessMaker/Helpers/SensitiveDataHelper.php [hunk 1]
```diff
@@ -5,8 +5,7 @@
 class SensitiveDataHelper
 {
     const SENSITIVE_KEYS = [
-        'password',
```
Comment (pmPaulis) [inline ProcessMaker/Helpers/SensitiveDataHelper.php:8]: Why you are removing those keys?

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Events/UserUpdated.php [hunk 2]
```diff
@@ -13,15 +14,23 @@ class UserUpdated implements SecurityLogEventInterface
...
      *
      * @return void
      */
-    public function __construct(User $user)
+    public function __construct(User $user, array $changes, array $original)
     {
         $this->user = $user;
+        $this->changes = $changes;
+        $this->original = $original;
+        unset($this->changes['meta']);
```
Comment (pmPaulis) [inline ProcessMaker/Events/UserUpdated.php:30]: Please define a list of Key to remove then remove using:
```
$this->changes = array_diff_key($changes, array_flip($this::REMOVE_KEYS));
$this->original = array_diff_key($original, array_flip($this::REMOVE_KEYS));
```

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Events/UserUpdated.php (hunk 2 above)
Comment (CarliPinell) [reply ProcessMaker/Events/UserUpdated.php:30]: Solved

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

File: ProcessMaker/Http/Controllers/Api/UserController.php [hunk 3]
```diff
@@ -296,16 +296,18 @@ public function update(User $user, Request $request)
...
         }
 
-        //Call new Event to store User Changes in LOG
-        UserUpdated::dispatch($user);
-
         $user->saveOrFail();
+        $changes = $user->getChanges();
+
+        //Call new Event to store User Changes into LOG
+        UserUpdated::dispatch($user, $changes,  $original);
```
Comment (luNunezProcessmaker) [inline ProcessMaker/Http/Controllers/Api/UserController.php:310]: Please delete extra space before $original

File: tests/unit/ProcessMaker/Helpers/ArrayHelperTest.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+use ProcessMaker\Helpers\ArrayHelper;
+use stdClass;
+use Tests\TestCase;
+
+class ArrayHelperTest extends TestCase
+{
+    public function testStdClassToArray()
+    {
+        $myObject = new stdClass();
+        $myObject->name = "John";
```
Comment (luNunezProcessmaker) [inline tests/unit/ProcessMaker/Helpers/ArrayHelperTest.php:14]: please use  single quotes

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

File: ProcessMaker/Http/Controllers/Api/UserController.php (hunk 3 above)
Comment (CarliPinell) [reply ProcessMaker/Http/Controllers/Api/UserController.php:310]: Solved

File: tests/unit/ProcessMaker/Helpers/ArrayHelperTest.php (hunk 4 above)
Comment (CarliPinell) [reply tests/unit/ProcessMaker/Helpers/ArrayHelperTest.php:14]: solved.

File: ProcessMaker/Helpers/ArrayHelper.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace ProcessMaker\Helpers;
+
+use stdClass;
```
Comment (pmPaulis) [inline ProcessMaker/Helpers/ArrayHelper.php:5]: Please apply alphabetical order

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

File: ProcessMaker/Helpers/ArrayHelper.php (hunk 5 above)
Comment (CarliPinell) [reply ProcessMaker/Helpers/ArrayHelper.php:5]: Solved.

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-2d5878ba0e.eng.processmaker.net

**Review by luNunezProcessmaker — APPROVED** [review]

**Review by pmPaulis — APPROVED** [review]

**Pull request merged**